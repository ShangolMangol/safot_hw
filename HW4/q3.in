val s = ref (new (counter ()));

print "===TEST START===\n";
s := next (!s);
s := next (!s);
s := prev (!s);
s := next (!s);
s := next (!s);
