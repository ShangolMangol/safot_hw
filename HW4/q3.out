Standard ML of New Jersey (64-bit) v110.99.2 [built: Sun Mar 20 20:10:23 2022]
[opening hw4_q3.sml]
[opening hw4_q3_def.sml]
datatype 'a seq = Cons of 'a * (unit -> 'a seq) | Nil
[autoloading]
[library $SMLNJ-BASIS/basis.cm is stable]
[library $SMLNJ-BASIS/(basis.cm):basis-common.cm is stable]
[autoloading done]
val counter = fn : unit -> int seq
val it = () : unit
type 'a biseq = int * 'a seq * (int * 'a seq) list
exception SeqErr
val new = fn : 'a seq -> int * 'a seq * (int * 'a seq) list
val curr = fn : 'a * 'b seq * 'c -> 'b
val empty = fn : 'a * 'b seq * 'c -> bool
val next = fn :
  int * 'a seq * (int * 'a seq) list -> int * 'a seq * (int * 'a seq) list
val prev = fn :
  int * 'a seq * (int * 'b seq) list -> int * 'b seq * (int * 'b seq) list
- val s = ref (0,Cons (0,fn),[(0,Cons (0,fn))]) :
  (int * int seq * (int * int seq) list) ref
===TEST START===
val it = () : unit
exec: 1
val it = () : unit
exec: 2
val it = () : unit
val it = () : unit
val it = () : unit
exec: 3
val it = () : unit
- 
