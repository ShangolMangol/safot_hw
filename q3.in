G
P
G
C
C
56
70
3
4
5
94
30
a
2
h
0
