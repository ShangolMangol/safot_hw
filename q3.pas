program TaskManager;
type Task = record 
    case letter: char of
        'P': (a,b,c : integer);
        'C': (ch : char; d : integer);
        'G': (average, exam: integer);
    end;

var tasks: array [1..5] of Task;
var i, j, finalGrade: integer;
var charInput: char;


begin
    for i := 1 to 5 do
    begin
        ReadLn(charInput);
        tasks[i].letter := charInput;
    end;

    for i := 1 to 5 do
    begin
        if tasks[i].letter = 'P' then
        begin
            ReadLn(tasks[i].a);
            ReadLn(tasks[i].b);
            ReadLn(tasks[i].c);
        end;
        if tasks[i].letter = 'C' then
        begin
            ReadLn(tasks[i].ch);
            ReadLn(tasks[i].d);
        end;
        if tasks[i].letter = 'G' then
        begin
            ReadLn(tasks[i].average);
            ReadLn(tasks[i].exam);
        end;
    end;

    for i := 5 downto 1 do
    begin
        if tasks[i].letter = 'P' then
        begin
            if tasks[i].a*tasks[i].a + tasks[i].b*tasks[i].b = tasks[i].c*tasks[i].c then
            begin
                writeLn('TRUE');
            end
            else
            begin
                writeLn('FALSE');
            end;
        end;
        if tasks[i].letter = 'C' then
        begin
            for j:=1 to tasks[i].d do
            begin
                tasks[i].ch := succ(tasks[i].ch);
            end;
            writeLn(tasks[i].ch);
        end;
        if tasks[i].letter = 'G' then
        begin
            if tasks[i].exam <= 54 then
            begin
                writeLn(tasks[i].exam);
            end
            else
            begin
                finalGrade := Round(tasks[i].exam*0.8 + tasks[i].average*0.2);
                writeLn(finalGrade);
            end;
        end;
    end;
end.