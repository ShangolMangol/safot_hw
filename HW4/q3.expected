val it = () : unit
exec: 1
val it = () : unit
exec: 2
val it = () : unit
val it = () : unit
val it = () : unit
exec: 3
val it = () : unit
- 
